django/db/models/indexes.py

```diff
@@ -0,0 +1,78 @@
+
+__all__ = ['Index']
+
+
+class BaseIndex(object):
+
+    # The type of index - BTree, Hash, RTree, etc.
+    # Used in deconstruction to differentiate indexes of different types with
+    # the same set of fields.
+    index_type = None
```

Comment:
It will also be used in the `describe` method in the Operations and other places afterwards. I'll keep in mind to see if all that could be done without introducing `index_type`.